### PR TITLE
Add APIs for IO mask editing

### PR DESCRIPTION
**Pull request by glutanimate**
Extends IO with a number of **JS and Python APIs** that facilitate interacting with the **mask editor** during adding or editing notes:

- Add `editor_mask_editor_did_load_image` hook that signals full initialization and image loading to Python
- Expose methods for querying, adding, and deleting shapes via `globalThis.maskEditor`

### CODE REVIEW

File: ts/image-occlusion/MaskEditor.svelte [hunk 1]
```diff
@@ -32,8 +34,17 @@ License: GNU AGPL, version 3 or later; http://www.gnu.org/licenses/agpl.html
...
 
+    $: {
+        globalThis.maskEditor = canvas ? new MaskEditorAPI(canvas) : null;
+    }
+
     const dispatch = createEventDispatcher();
 
+    function onImageLoaded() {
+        const data = mode.kind === "add" ? mode.imagePath : mode.noteId;
+        bridgeCommand(`ioImageLoaded:${JSON.stringify(data)}`);
```
Comment (krmanik) [inline ts/image-occlusion/MaskEditor.svelte:45]: This line may not work on Mobile client because this is specific desktop client.

Comment (glutanimate) [reply ts/image-occlusion/MaskEditor.svelte:45]: Thanks for taking a look, Mani!

Would this raise an exception in this case, or is `bridgeCommand` just a no-op on mobile? I was assuming the latter, but don't actually know for sure.

Comment (krmanik) [reply ts/image-occlusion/MaskEditor.svelte:45]: This will be the error.
```
Uncaught ReferenceError: bridgeCommand is not defined.
```

Comment (glutanimate) [reply ts/image-occlusion/MaskEditor.svelte]: Thanks! I moved the `bridgeCommand` calls to `NoteEditor`, which should fix the mobile compat hopefully.

Comment (dae) [reply ts/image-occlusion/MaskEditor.svelte]: Good catch Mani!

**Pull request merged**